### ISSUE TITLE
The linearize() API crashes OMC 1.19.0

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#8416]
### Description

Running the linearize() API from either script or OMEdit CLI crashes the current development version of 1.19.0. It worked well in 1.18.0. I have no idea when this actually broke.

### Steps to Reproduce

Run the following test script:
```
loadModel(Modelica);getErrorString();
linearize(Modelica.Blocks.Examples.Filter, stopTime = 1);getErrorString();
```
OMC dies abruptly with no error message.

The same happens if MSL is loaded in OMEdit and the linearize command is issued in the CLI interface.

### Expected Behavior

The linearized model should be dumped to the working directory.

### Additional Context

I would suggest to add the above script to the core testuite, so we make sure to catch bugs that break linearization in the future.

**Comment by mahge** [comment]
The model works fine for me from the command line. 

```
simulationOptions = "startTime = 0.0, stopTime = 1.0, numberOfIntervals = 500, tolerance = 1e-06, method = 'dassl', fileNamePrefix = 'Modelica.Blocks.Examples.Filter', options = '', outputFormat = 'mat', variableFilter = '.*', cflags = '', simflags = ''",
    messages = "stdout            | info    | Linearization will be performed at point of time: 1.000000
LOG_SUCCESS       | info    | The initialization finished successfully without homotopy method.
LOG_SUCCESS       | info    | The simulation finished successfully.
stdout            | info    | Linear model is created at /home/mahder/dev/OpenModelica/working/ticket_8415/linearized_model.mo
stdout            | info    | The output format can be changed with the command line option --linearizationDumpLanguage.
stdout            | info    | The options are: --linearizationDumpLanguage=modelica, matlab, julia, python.
",
    timeFrontend = 0.171229249,
    timeBackend = 0.034779888,
    timeSimCode = 0.010335162,
    timeTemplates = 0.031365277,
    timeCompile = 0.541516977,
    timeSimulation = 0.009536541000000001,
    timeTotal = 0.798886489
end SimulationResult;
```

But as you said OMEdit crashes when it is issued in the OMEdit CLI. So this is probably an OMEdit issue.

**Comment by casella** [comment]
> The model works fine for me from the command line.

What I did was to save those two lines in a `test.mos` file and then run `omc test.mos`. Is that what you did?

**Comment by mahge** [comment]
> What I did was to save those two lines in a test.mos file and then run omc test.mos. Is that what you did?

Yes exactly. It works in that case.

The problem turned out to be the `stopTime = 1` part. More specifically the `=` not being handled by the new flat Modelica expression parser we have now.  

The flat Modelica expression parser is designed to parse expressions which it does properly. Except a call expression is expected to be parsed as `CALL ( EXP {, EXP} )`.

In this case we get a something like a `CALL ( IDENT, #error#)` because `stopTime = 1` is not an actually valid expression. It is a _named argument_  which needs to be treated specifically when it is part of a call expression.  Since it can not handle it, the parser throws an exception which is not caught by OMEdit. Resulting in OMEdit crashing.

Why is the expression parser even called here? 
OMEdit wants to know when some specific commands (e.g. _loadLibrary()_) are issued through the CLI in order to refresh the library browser. See #8100. For this it uses the expression parser to parse the command. So if the parser throws OMEdit just crashes since it does not try to catch anything. 

Right now if you type any invalid expression on the CLI OMEdit will just crash.   

The parser is probably being used for purposes that it was not originally intended for. It might be a good idea to fix it nonetheless. However, the most pressing issue is to improve the interaction between OMEdit and the parser. Either OMEdit needs to catch the expectations thrown by the parser and deal with them properly (I think this is the way to go) or the parser should stop throwing. 

@perost @adeas31.

**Comment by casella** [comment]
> > What I did was to save those two lines in a test.mos file and then run omc test.mos. Is that what you did?
> 
> Yes exactly. It works in that case.

OK, works for me as well now. Not sure what I did earlier.

**Comment by casella** [comment]
> The problem turned out to be the `stopTime = 1` part. More specifically the `=` not being handled by the new flat Modelica expression parser we have now.

Would it be possible to use the previous one for this purpose? That worked well.

> Right now if you type any invalid expression on the CLI OMEdit will just crash.

This is obviously not good. We need to catch those exceptions and print out some diagnostic message.

> The parser is probably being used for purposes that it was not originally intended for. It might be a good idea to fix it nonetheless. However, the most pressing issue is to improve the interaction between OMEdit and the parser. Either OMEdit needs to catch the exceptions thrown by the parser and deal with them properly (I think this is the way to go) or the parser should stop throwing.

See above.

In fact, this specific issue could also be handled by fixing [#5626](https://trac.openmodelica.org/OpenModelica/ticket/5626). @adeas31, this looks quite simple, could you do that?

**Comment by mahge** [comment]
This has been fixed as part of #8429.

**Comment by casella** [comment]
Confirmed, thanks!
